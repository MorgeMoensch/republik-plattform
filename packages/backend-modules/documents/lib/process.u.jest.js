jest.mock('@orbiting/backend-modules-auth', () => ({
  Roles: {
    userIsInRoles: jest.fn(),
  },
}))

const UUT = require('./process')

const getMdast = () => ({
  identifier: 'CENTER',
  data: {},
  children: [
    {
      children: [
        {
          type: 'text',
          value: 'Hello',
        },
      ],
      type: 'paragraph',
    },
    {
      identifier: 'IF',
      data: {
        present: 'hasAccess',
      },
      children: [
        {
          children: [
            {
              type: 'text',
              value: 'This ',
            },
            {
              children: [
                {
                  type: 'text',
                  value: 'part',
                },
              ],
              type: 'emphasis',
            },
            {
              type: 'text',
              value: ' is visible to ',
            },
            {
              children: [
                {
                  type: 'text',
                  value: 'users w/ access',
                },
              ],
              type: 'strong',
            },
            {
              type: 'text',
              value: '.',
            },
          ],
          type: 'paragraph',
        },
        {
          identifier: 'ELSE',
          data: {},
          children: [
            {
              children: [
                {
                  children: [
                    {
                      type: 'text',
                      value: 'Users lacking access, will see this.',
                    },
                  ],
                  type: 'strong',
                },
              ],
              type: 'paragraph',
            },
          ],
          type: 'zone',
        },
      ],
      type: 'zone',
    },
    {
      children: [
        {
          type: 'text',
          value: 'In between.',
        },
      ],
      type: 'paragraph',
    },
    {
      identifier: 'IF',
      data: {
        present: 'hasAccess',
      },
      children: [
        {
          children: [
            {
              type: 'text',
              value: 'Block only shows IF hasAccess === true',
            },
          ],
          type: 'paragraph',
        },
      ],
      type: 'zone',
    },
    {
      children: [
        {
          type: 'text',
          value: 'In between again.',
        },
      ],
      type: 'paragraph',
    },
    {
      identifier: 'IF',
      data: {
        present: 'hasAccess',
      },
      children: [
        {
          identifier: 'ELSE',
          data: {},
          children: [
            {
              children: [
                {
                  type: 'text',
                  value:
                    'Block visible to users w/o access in an empty IF hasAccess block.',
                },
              ],
              type: 'paragraph',
            },
          ],
          type: 'zone',
        },
      ],
      type: 'zone',
    },
    {
      children: [
        {
          type: 'text',
          value: 'Live long, and prosper.',
        },
      ],
      type: 'paragraph',
    },
  ],
  type: 'zone',
})

describe('process', () => {
  describe('processIfHasAccess', () => {
    it('keeps non-member nodes if user has no access', () => {
      const { Roles } = require('@orbiting/backend-modules-auth')
      Roles.userIsInRoles.mockReturnValue(false)

      const mdast = getMdast()

      UUT.processIfHasAccess(mdast)
      expect(mdast).toMatchSnapshot()
    })

    it('keeps member nodes if user has access', () => {
      const { Roles } = require('@orbiting/backend-modules-auth')
      Roles.userIsInRoles.mockReturnValue(true)

      const mdast = getMdast()
      UUT.processIfHasAccess(mdast)
      expect(mdast).toMatchSnapshot()
    })
  })
})
