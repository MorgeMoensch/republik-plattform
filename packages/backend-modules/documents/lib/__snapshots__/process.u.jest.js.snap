// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process processIfHasAccess keeps member nodes if user has access 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Hello",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "This ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "part",
            },
          ],
          "type": "emphasis",
        },
        Object {
          "type": "text",
          "value": " is visible to ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "users w/ access",
            },
          ],
          "type": "strong",
        },
        Object {
          "type": "text",
          "value": ".",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Block only shows IF hasAccess === true",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between again.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Live long, and prosper.",
        },
      ],
      "type": "paragraph",
    },
  ],
  "data": Object {},
  "identifier": "CENTER",
  "type": "zone",
}
`;

exports[`process processIfHasAccess keeps non-member nodes if user has no access 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Hello",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Users lacking access, will see this.",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between again.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Block visible to users w/o access in an empty IF hasAccess block.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Live long, and prosper.",
        },
      ],
      "type": "paragraph",
    },
  ],
  "data": Object {},
  "identifier": "CENTER",
  "type": "zone",
}
`;

exports[`process processIfHasAccess removes member parts if user has no access 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Hello",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "Users lacking access, will see this.",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between again.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Block visible to users w/o access in an empty IF hasAccess block.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Live long, and prosper.",
        },
      ],
      "type": "paragraph",
    },
  ],
  "data": Object {},
  "identifier": "CENTER",
  "type": "zone",
}
`;

exports[`process processIfHasAccess removes non-member parts if user has access 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Hello",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "This ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "part",
            },
          ],
          "type": "emphasis",
        },
        Object {
          "type": "text",
          "value": " is visible to ",
        },
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "users w/ access",
            },
          ],
          "type": "strong",
        },
        Object {
          "type": "text",
          "value": ".",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Block only shows IF hasAccess === true",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "In between again.",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Live long, and prosper.",
        },
      ],
      "type": "paragraph",
    },
  ],
  "data": Object {},
  "identifier": "CENTER",
  "type": "zone",
}
`;
